port/osx/omrvmem.c

```diff
@@ -454,12 +462,46 @@ reserveMemory(struct OMRPortLibrary *portLibrary, void *address, uintptr_t byteA
 		fd = VM_FLAGS_SUPERPAGE_SIZE_ANY;
 	}
 
+	/* We only create shared heap if it was requested and huge pages are not available */
+	if ((-1 == fd) && useBackingSharedFile) {
+		Trc_PRT_vmem_reserve_doubleMapAPI_available(address, byteAmount);
+		mode |= OMRPORT_VMEM_MEMORY_MODE_DOUBLE_MAP_AVAILABLE;
+		int ft = -1;
+		/* shm_open(2) only accepts file names up to SHM_NAME_MAX (32) chars in size (including the
+		 * NULL character). To avoid overflow, we truncate omrtime_current_time_str to 12 digits
+		 * and pthread_self to 9 digits (totatlling 30 chars plus NULL char), which is enough for
+		 * the purposes of generating a random name. */
+		char omrtime_current_time_str[FILE_NAME_SIZE];
+		char pthread_self_str[FILE_NAME_SIZE];
+		char filename[FILE_NAME_SIZE];
+		sprintf(omrtime_current_time_str, "%zu", (uintptr_t)omrtime_current_time_millis(portLibrary));
+		sprintf(pthread_self_str, "%zu", (uintptr_t)pthread_self());
+		sprintf(filename, "omrvmem_%.12s_%.9s", omrtime_current_time_str, pthread_self_str);
```

Comment:
Seems the last 12 would be more unique than the first 12 digits.

Reply:
I totally agree but there’s not an easy way to truncate from the right side. One option I found was:
```
sprintf(omrtime_current_time_str, "%zu", (uintptr_t)omrtime_current_time_millis(portLibrary) % (uintptr_t)pow(10,12));
sprintf(pthread_self_str, "%zu", (uintptr_t)pthread_self() % (uintptr_t)pow(10,9));
sprintf(filename, "omrvmem_%.12s_%.9s", omrtime_current_time_str, pthread_self_str);
```
But I didn’t want to add extra arithmetics (in this case it would be 4) to just name a variable. Maybe I’m overthinking and it’s okay?

pow(10,12) is actually a constant so it could become  `(uintptr_t)omrtime_current_time_millis(portLibrary) % 1000000000000`

Reply:
Seems fine using constant values.